proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,456 @@
+finalized, the proposal was made public for open comments and to get feedback
+that the approach would solve the problems that exist. The change was a major
+deviation from how the AWS CLI works currently, and it was felt that having a
+formalized process for this change would result in better decisions.
+
+In this case, we would use the same format as this document for a proposal that
+would be opened for public comment and review. The document would be
+incorporated into the codebase as an accepted proposal as a matter of record.
+
+### Q. Why should there be a minimum number of upvotes or reactions for a feature request?
+
+Feature requests carry a risk of adding new technical debt and increased
+maintenance cost and should have a higher bar to be included.
+
+The request should have user impact beyond the individual who proposed it. This
+is difficult to measure objectively, but one proxy for impact can be the number
+of upvote reactions on the issue. This is intended to prevent narrowly scoped
+changes that only affect a single user. It also helps to limit the number of in
+flight reviews that the maintainers need to be engaged on, as there is not an
+unlimited amount of bandwidth to devote to this task.
+
+At present, there are more than 95 open issues with at least 10 upvotes. We will
+use the current upvote counts to review the backlog and identify issues that we
+feel would be suited for community contributions. This process can then inform
+us how to handle new issues that might be candidates for community contribution.
+
+As part of the contribution guide, we will provide more specific guidance on how
+to upvote a feature so we can more accurately estimate user impact.
+
+### Q. Do users need to request or acknowledge to work on an issue that is ready for contribution?
+
+No. We do not want to require that users who want to contribute pre-register
+their interest. Multiple contributions can result in a better implementation.
+However, we should encourage users to comment on the issue that they are working
+on it. This can be an opportunity for community collaboration instead of
+duplicated effort.
+
+### Q. Why are issue types prioritized in the order of bug fixes, documentation, and feature requests?
+
+Once a bug has been triaged and confirmed, it should be made available for
+community contribution immediately without any upvote requirements. Bug fixes
+have an immediate impact on affected users from using the AWS CLI as intended.
+
+Most documentation changes are low risk and should move quickly through the
+review process. Documentation changes also have a high reward potential,
+measured by the number of users who would read the documentation to solve their
+problem. In addition, there are also more maintainers available for validation
+and review of documentation changes.
+
+Feature requests need to be carefully assessed before taking on new potential
+technical debt.
+
```

Comment:
Other questions, that I think we should add:

1. Why do maintainers complete the pull request in the review phase once it becomes stale?
2. Why introduce a GitHub Kanban board?
3. What if it is easier to express how to go about a change via a pull request during the intake stage?